src/ElfReader.cc

```diff
@@ -5,12 +5,43 @@
 #include <elf.h>
 #include <sys/mman.h>
 #include <sys/stat.h>
+#include <zlib.h>
 
 #include "log.h"
 #include "util.h"
 
 using namespace std;
 
+namespace {
+
+struct DecompressedSection {
+  uint64_t compressed_start;
+  uint8_t* start;
```

Comment:
Document that this points to an mmapped region we own. But, why mmap instead of say just `std::vector<uint8_t>`?